gobblin-data-management/src/main/java/gobblin/data/management/copy/recovery/RecoveryHelper.java

```diff
@@ -119,6 +129,26 @@ public boolean persistFile(State state, CopyableFile file, Path path) throws IOE
   }
 
   /**
+   * Delete all persisted files older than the number of hours set by {@link #PERSIST_RETENTION_KEY}.
+   * @throws IOException
+   */
+  public void purgeOldPersistedFile() throws IOException {
+    if (!this.persistDir.isPresent() || !this.fs.exists(this.persistDir.get())) {
+      log.info("No persist directory to clean.");
+      return;
+    }
+
+    long retentionMillis = TimeUnit.HOURS.toMillis(this.retentionHours);
+    long now = System.currentTimeMillis();
+
+    for (FileStatus fileStatus : this.fs.listStatus(this.persistDir.get())) {
+      if (now - fileStatus.getModificationTime() > retentionMillis) {
+        this.fs.delete(fileStatus.getPath(), true);
```

Comment:
minor: should we log if delete() returns false - could happen if somehow permissions mess up or if two executions have a run away condition (though it has a very small probability )